libslab/app-shell.c

```diff
@@ -265,9 +265,19 @@ layout_shell (AppShellData * app_data, const gchar * filter_title, const gchar *
 	right_vbox = gtk_box_new (GTK_ORIENTATION_VERTICAL, 0);
 
 	num_cols = SIZING_SCREEN_WIDTH_LARGE_NUMCOLS;
-	if (WidthOfScreen (gdk_x11_screen_get_xscreen (gdk_screen_get_default ())) <= SIZING_SCREEN_WIDTH_LARGE)
+
+	GdkRectangle geometry = {0};
+
+	GdkDisplay *display;
+	GdkMonitor *monitor;
+
+	display= gdk_screen_get_display (gdk_screen_get_default ());
+	monitor = gdk_display_get_monitor (display, 0);
```

Comment:
No testing, testing later. 
My question is:
Can we still work properly when we have multiple monitors ？

Reply:
About to find out right now: have dissimilar window scaling monitors on wayland, and same scaling on xorg. If we find a bug we can split this change out to be wayland-only, but the only likely issue would be on dissimilar monitors as all this does is decide how much space is available for columns.

Reply:
Works quite nice on wayland even with a 4K and 1080p monitor side by side, though the scaled size is exactly the same. Will compare on Xorg to git master now

Reply:
Dissimilar multimonitor test on X11 gave exact same results as on master: control center fullscreened, overflows off bottom (NOT in width) off smaller monitor when 1080p secondary monitor is running at window-scaling =  2 because dissimilar window scaling via xrandr in Xorg is so slow. Horizontally it was resized to exact fit, and on the larger monitor no observable difference in Xorg between this and git master.

Reply:
In the old days, monitors could range from perhaps 1600 x 1200 all the way down to 640 x 480 or more reasonably 800 x 600. As late as 2011 netbooks were shipping with 1024 x 600 screens. Now we see higher dpi on even the smaller screens. Ratio of smallest to largest monitors hasn't changed much, but absolute resolution has.

Reply:
From https://docs.gtk.org/gdk3/method.Display.get_monitor.html
Return value `The GdkMonitor, or NULL if monitor_num is not a valid monitor number`
Guessing that `0` isn't a valid `monitor_num` than `Null` is the return value.
So this change seems to be OK, ...or not?
I am not complete sure what zero means for num_monitor...., description in docs is very minimal.

Reply:
0 would be the first monitor as determined by the video card's firmware and the video driver when multiple monitors are connected. I suspect this number goes to the first monitor the card detects, but not sure of that. Subsequent monitors would be numbered 1, 2, etc. Thus there should always be a monitor # 0 unless running headless or that monitor has been disconnected after startup in favor of another. 

from https://docs.gtk.org/gdk3/method.Display.get_monitor.html
```
monitor_num

    Type: int

    Number of the monitor.
```